enhancements/local-storage/automatic-detection.md

```diff
@@ -0,0 +1,458 @@
+	NodeName string `json:"nodeName"`
+	// If specified tolerations is the list of toleration that is passed to the
+	// LocalVolumeDiscovery Daemon
+	// +optional
+	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
+}
+
+type NodeDeviceMapStatus struct {
+	Phase      string `json:"phase,omitempty"`
+	TimeStramp string `json:"timestramp,omitempty"`
+	// Blockdevices will be containing the output of following command unmarshalled into the structured
+	// "lsblk -o name,model,vendor,kname,pkname,type,uuid,tran,ro,rm,parttype,serial,size,rota,mountpoint"
+	Blockdevices []Device `json:"blockdevices"`
+}
+
+type Device struct {
+	Name       string   `json:"name,omitempty"`
+	Model      string   `json:"model,omitempty"`
+	Vendor     string   `json:"vendor,omitempty"`
+	Kname      string   `json:"kname,omitempty"`
+	Pkname     string   `json:"pkname,omitempty"`
+	Type       string   `json:"type,omitempty"`
+	UUID       string   `json:"uuid,omitempty"`
+	Tran       string   `json:"tran,omitempty"`
+	Ro         string   `json:"ro,omitempty"`
+	Parttype   string   `json:"parttype,omitempty"`
+	Serial     string   `json:"serial,omitempty"`
+	Size       string   `json:"size,omitempty"`
+	Mountpoint string   `json:"mountpoint,omitempty"`
+	Rm         string   `json:"rm,omitempty"`
+	Fstype     string   `json:"fstype,omitempty"`
+	Children   []Device `json:children,omitempty`
+}
+```
+
+### Workflow of `Auto Device Discovery`
+The discovery of local devices will be done by the `LocalVolumeDiscovery` and `LocalVolumeNodeDiscovery` CR. 
+#### LocalVolumeDiscovery CR
+- Once the `LocalVolumeDiscovery` CR is deployed, the controller of the `LocalVolumeDiscovery` CR will be managing `LocalVolumeNodeDiscovery` CR for each node that matches the `nodeSelector`.
+- Possible `status.phase` for `LocalVolumeDiscovery` CR:
+  - `Deploying`
+  - `Failed`
+  - `Deployed`
+Example of `LocalVolumeDiscovery` CR:
+```yaml
+apiVersion: local.storage.openshift.io/v1alpha1
+kind: LocalVolumeDiscovery
+metadata:
+  name: example-autodetect
+  namespace: localStorage
+spec:
+  nodeSelector:
+    nodeSelectorTerms:
+      - matchExpressions:
+          - key: kubernetes.io/hostname
+            operator: In
+            values:
+              - worker-0
+              - worker-1
+status:
+  phase: Deployed
+```
+
+#### LocalVolumeNodeDiscovery CR
+- The `LocalVolumeNodeDiscovery` CR will be exposing the available local devices of the node in its status.
+- These information will mostly be the output of  `lsblk -o name,model,vendor,kname,pkname,fstype,type,uuid,tran,ro,rm,parttype,serial,size,rota,mountpoint`.
+- The controller for `LocalVolumeNodeDiscovery` CR will be deploying a discovery job for the node to which the CR refers.
+- Possible `status.phase` for `LocalVolumeNodeDiscovery` CR:
+  - `Discovering`
+  - `Failed`
+  - `Discovered`
+- If the CR for `LocalVolumeNodeDiscovery` CR already exits then updating the CR will retriger the discovery process on that node.
```

Comment:
Is this intended to reflect daemonset rollout status?

Reply:
Nope The discovery will be done via job not via daemonset